src/collectors/jolokia/kafka_jolokia.py

```diff
@@ -0,0 +1,592 @@
+"""
+Collects Kafka JMX metrics from the Jolokia Agent.  Extends the
+JolokiaCollector to reformat the Kafka MBean names into easier to
+understand metric names.
+
+Additionally, we also connect to Zookeeper to collect certain 
+metrics that are critical to Kafka monitoring.  This is done in
+two ways:
+    1) By extending the ProcessCollector, we can execute a command-
+        line function against the Zookeeper instance to extract 
+        infomation about the consumer groups and their corresponding
+        consumer lag.  This replaces and improves upon the 
+        KafkaConsumerLagCollector.
+    2) By extending the ZookeeperCollector, we can collect the 
+        key Zookeeper metrics for Kafka without requiring the user
+        to configure a separate collector. This does not replace
+        the Zookeeper collector, however; users with a Zookeeper
+        instance being used for something other than Kafka can still
+        use the Zookeeper collector on its own. 
+
+Netuitive Change History
+    2016/10/26 DVG - Initial version.
+    2016/11/16 DVG - Added debug logging, additional error handling, refactoring of common code across the Kafka 8 and 9 handling.
+                     Fixed bug with Zookeeper collector defaulting to localhost:2181
+"""
+
+from jolokia import JolokiaCollector
+from zookeeper import ZookeeperCollector
+from diamond.collector import ProcessCollector
```

Comment:
Given the complexity of the collector model and the state of automated testing, I'd much prefer a little duplicate code rather than a web of dependancy.